pkg/schedule/schedulers/balance_leader.go

```diff
@@ -57,13 +57,14 @@ const (
 
 var (
 	// WithLabelValues is a heavy operation, define variable to avoid call it every time.
-	balanceLeaderScheduleCounter         = schedulerCounter.WithLabelValues(BalanceLeaderName, "schedule")
-	balanceLeaderNoLeaderRegionCounter   = schedulerCounter.WithLabelValues(BalanceLeaderName, "no-leader-region")
-	balanceLeaderRegionHotCounter        = schedulerCounter.WithLabelValues(BalanceLeaderName, "region-hot")
-	balanceLeaderNoTargetStoreCounter    = schedulerCounter.WithLabelValues(BalanceLeaderName, "no-target-store")
-	balanceLeaderNoFollowerRegionCounter = schedulerCounter.WithLabelValues(BalanceLeaderName, "no-follower-region")
-	balanceLeaderSkipCounter             = schedulerCounter.WithLabelValues(BalanceLeaderName, "skip")
-	balanceLeaderNewOpCounter            = schedulerCounter.WithLabelValues(BalanceLeaderName, "new-operator")
+	balanceLeaderScheduleCounter           = schedulerCounter.WithLabelValues(BalanceLeaderName, "schedule")
+	balanceLeaderNoLeaderRegionCounter     = schedulerCounter.WithLabelValues(BalanceLeaderName, "no-leader-region")
+	balanceLeaderRegionHotCounter          = schedulerCounter.WithLabelValues(BalanceLeaderName, "region-hot")
+	balanceLeaderNoTargetStoreCounter      = schedulerCounter.WithLabelValues(BalanceLeaderName, "no-target-store")
+	balanceLeaderNoFollowerRegionCounter   = schedulerCounter.WithLabelValues(BalanceLeaderName, "no-follower-region")
+	balanceLeaderSkipCounter               = schedulerCounter.WithLabelValues(BalanceLeaderName, "skip")
+	balanceLeaderNewOpCounter              = schedulerCounter.WithLabelValues(BalanceLeaderName, "new-operator")
+	balanceLeaderStoreRecentlySplitRegions = schedulerCounter.WithLabelValues(BalanceLeaderName, "source-recently-split-regions")
```

Comment:
I think the metrics are a bit confusing here, when the source store is filtered, it is counted in 'filterCounter', but in 'transferLeaderIn', it is counted in 'schedulerCounter'.  And the label name are also different.

Reply:
fixed by add region filter to filter out the no splitted store.